packages/api/AlvTime.Business/Customers/CustomerService.cs

```diff
@@ -5,39 +5,43 @@
 
 namespace AlvTime.Business.Customers
 {
-    public class CustomerCreator
+    public class CustomerService
     {
         private readonly ICustomerStorage _storage;
 
-        public CustomerCreator(ICustomerStorage storage)
+        public CustomerService(ICustomerStorage storage)
         {
             _storage = storage;
         }
 
         public CustomerDto CreateCustomer(CustomerDto customer)
         {
-            if (!GetCustomer(customer).Any())
+            CustomerQuerySearch criterias = new CustomerQuerySearch
+            {
+                Name = customer.Name,
+                Id = customer.Id
+            };
+            if (!GetCustomers(criterias).Any())
```

Comment:
Dette er helt frivillig om du vil gjøre, men greit å vite om uansett. Vi har en ErrorHandlingMiddleware.cs som fanger Exceptions og outputter de i et standard format som frontend kan vise i en pen feilmelding. Her er en case hvor det kanskje ville gitt mening å kastet et exception dersom man prøver å lage en kunde som allerede finnes. I ErrorHandlingMiddleware.cs har vi mulighet til å spesifisere hva slags statuskode en viss type exception vil gi. Her kunne vi for eksempel kastet et ValidationException som da vil gi brukeren en 400 Bad Request. Det ville da sett ut noe som dette:
`if (GetCustomers(criterias).Any(){
  throw new ValidationException("Denne kunden eksisterer allerede");
}
_storage.CreateCustomer(customer);`
Dette ville også vært en bra case å skrive en enhetstest for :)

Reply:
Prøvde å implementere dette, men fikk en feilmelding. Først antok jeg at jeg bare måtte referere til `ErrorHandlingMiddleware.cs`, men det kom ikke opp som et alternativ

![image](https://user-images.githubusercontent.com/47418059/161432846-e0d928fc-b2d0-4bdc-a6c9-721497380e07.png)

Reply:
Bare velg den nederste der: `using FluentValidation` så går det nok fint 😄 Ser også en skrivefeil der i `_cusomterStorage`.